src/resonate/scheduler.py

```diff
@@ -558,67 +739,53 @@ def _run(self) -> None:  # noqa: C901, PLR0912
 
             assert not self._runnable_coros, "Runnables should have been all exhausted"
 
-    def _add_coro_to_awaitables(
-        self, p: Promise[Any], coro_and_promise: CoroAndPromise[Any]
-    ) -> None:
+    def _add_coro_to_awaitables(self, p: Promise[Any], coro: ResonateCoro[Any]) -> None:
         assert (
             not p.done()
         ), "If the promise is resolved already it makes no sense to block coroutine"
-        self._awaitables.setdefault(p, []).append(coro_and_promise)
-        coro_and_promise.route_info.promise.increase_blocked_on()
+        self._awaitables.setdefault(p, []).append(coro)
+
+        if isinstance(p.action, LFI) and isinstance(p.action.exec_unit, Command):
+            assert not isinstance(
+                p.action.exec_unit, CreateDurablePromiseReq
+            ), "This command is reserved for rfi."
+            self._send_pending_commands_to_processor(type(p.action.exec_unit))
+
+        coro.route_info.promise.increase_blocked_on()
```

Comment:
Same as the other comment, let's check if we still need this code.

Reply:
Remvoed